oauthproxy.go

```diff
@@ -609,10 +612,46 @@ func (p *OAuthProxy) IsWhitelistedPath(path string) bool {
 	return false
 }
 
-func getRemoteAddr(req *http.Request) (s string) {
+// Returns (nil, nil) if no client IP found matching requirements found (missing header).
+func (p *OAuthProxy) GetRealClientIP(req *http.Request) (*net.IP, error) {
+	var ipStr string
+	if p.realClientIPHeader != "" {
+		if realIP := req.Header.Get(p.realClientIPHeader); realIP != "" {
+			ipStr = realIP
+		} else {
+			return nil, nil
+		}
+	} else {
+		splitIP, _, err := net.SplitHostPort(req.RemoteAddr)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to address (%s) from http.RemoteAddr", req.RemoteAddr)
+		}
+		ipStr = splitIP
+	}
+
+	if commaIndex := strings.IndexRune(ipStr, ','); commaIndex != -1 {
+		ipStr = ipStr[:commaIndex]
+	}
+	ipStr = strings.TrimSpace(ipStr)
+
+	if ipHost, _, err := net.SplitHostPort(ipStr); err == nil {
+		ipStr = ipHost
+	}
+
+	ip := net.ParseIP(ipStr)
+	if ip == nil {
+		return nil, fmt.Errorf("Unable to parse IP (%s) from %s header", ipStr, p.realClientIPHeader)
+	}
+
+	return &ip, nil
+}
+
+func (p *OAuthProxy) GetRemoteAddr(req *http.Request) (s string) {
```

Comment:
I'm not sure `GetRemoteAddr` is the right name for this method.  I kept it as it's essentially an upgraded version of the removed `getRemoteAddr` function.  But I'm wondering if there isn't a better way to name this.